src/lib.rs

```diff
@@ -1876,37 +1876,11 @@ fn layers(
     Ok(PyList::new(py, output).into())
 }
 
-/// Get the distance matrix for a directed graph
-///
-/// This differs from functions like digraph_floyd_warshall_numpy in that the
-/// edge weight/data payload is not used and each edge is treated as a
-/// distance of 1.
-///
-/// This function is also multithreaded and will run in parallel if the number
-/// of nodes in the graph is above the value of ``parallel_threshold`` (it
-/// defaults to 300). If the function will be running in parallel the env var
-/// ``RAYON_NUM_THREADS`` can be used to adjust how many threads will be used.
-///
-/// :param PyDiGraph graph: The graph to get the distance matrix for
-/// :param int parallel_threshold: The number of nodes to calculate the
-///     the distance matrix in parallel at. It defaults to 300, but this can
-///     be tuned
-/// :param bool as_undirected: If set to ``True`` the input directed graph
-///     will be treat as if each edge was bidirectional/undirected in the
-///     output distance matrix.
-///
-/// :returns: The distance matrix
-/// :rtype: numpy.ndarray
-#[pyfunction(parallel_threshold = "300", as_undirected = "false")]
-#[pyo3(
-    text_signature = "(graph, /, parallel_threshold=300, as_undirected=False)"
-)]
-pub fn digraph_distance_matrix(
-    py: Python,
+fn directed_distance_matrix(
     graph: &digraph::PyDiGraph,
     parallel_threshold: usize,
     as_undirected: bool,
-) -> PyResult<PyObject> {
+) -> Array2<f64> {
```

Comment:
This is an old commit but in my opinion it shows a general weakness of our source code: most of our functions are tightly connected to Python and it's not straight forward to re-use them inside Rust. As we expand retworkx's functionality i guess we will hit such cases more often.

Reply:
Yeah, that's a good point and I agree with you. But it's kind of understandable as most of the code is designed with python in mind first and reuse in rust is secondary. I do hope having the new organizational structure in the repo will make us a bit better about writing things for reuse since it gives us a clear divide on the boundary between the python interface side and the internal rust functions.

FWIW, I did end up doing this as part of #405 as a separate change since it wasn't needed here anymore.